{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d207b4bc",
   "metadata": {},
   "source": [
    "# Understanding the python client libraries generated using `openapi-python-client`\n",
    "\n",
    "<details>\n",
    "<summary style=color:green> \n",
    "As stated in the README:\n",
    "</summary>\n",
    "<h1></h1>\n",
    "\n",
    "Every path/method combo becomes a Python module with four functions:\n",
    "1. `sync`: Blocking request that returns parsed data (if successful) or `None`\n",
    "2. `sync_detailed`: Blocking request that always returns a `Request`, optionally with `parsed` set if the request was successful.\n",
    "3.  `asyncio`: Like `sync` but async instead of blocking\n",
    "4. `asyncio_detailed`: Like `sync_detailed` but async instead of blocking\n",
    "\n",
    "- All path/query params, and bodies become method arguments.\n",
    "- If your endpoint had any tags on it, the first tag will be used as a module name for the function (my_tag above)\n",
    "\n",
    "<h1></h1>\n",
    "</details>\n",
    "<br>\n",
    "\n",
    "Below we will have an example where we will get all studies of the biome and their sample counts: Root:Host-associated:Plants:Rhizosphere\n",
    "\n",
    "-----"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ddcde1ce",
   "metadata": {},
   "source": [
    "At a minimum:\n",
    "\n",
    "1. We need to find the appropriate module for our query: `studies/studies_list.py`\n",
    "2. Provide the `mgni_py.Client` instance\n",
    "3. (Optional) add type annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ba92758e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# at minimum need\n",
    "# 1. the path \n",
    "from mgni_py_v1.api.studies import studies_list\n",
    "# 2. the client\n",
    "from mgni_py_v1 import Client as ClientOne\n",
    "# extra nice to have annotations\n",
    "# 3. the models\n",
    "from mgni_py_v1.models import PaginatedStudyList\n",
    "from mgni_py_v1.types import Response, UNSET"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e260f82e",
   "metadata": {},
   "source": [
    "To instantiate the python client we really only need the base_url. However there are options for loggiing and other httpx args. \n",
    "\n",
    "`mgni_py_v1.Client` will take care of constructing and closing the httpx clients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d3f7e63",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Client(raise_on_unexpected_status=False, _base_url='https://www.ebi.ac.uk/metagenomics/api/', _cookies={}, _headers={}, _timeout=None, _verify_ssl=True, _follow_redirects=False, _httpx_args={}, _client=None, _async_client=None)\n"
     ]
    }
   ],
   "source": [
    "example_client = ClientOne(\n",
    "    base_url = \"https://www.ebi.ac.uk/metagenomics/api/\"\n",
    ")\n",
    "# check it out\n",
    "print(example_client)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31288451",
   "metadata": {},
   "source": [
    "The get request is made when running the `.sync...()` or `.async...()` functions. For example if executing\n",
    "```python\n",
    "with example_client as client:\n",
    "    response = studies_list.sync_detailed(\n",
    "        client=client, page_size=10\n",
    "    )\n",
    "```\n",
    "the order of the methods within `studies_list.py` is:\n",
    "\n",
    "1. `_get_kwargs` to prepare the query params, ensuring that the kwarg exists e.g. that `page_size` is an acceptable kwarg\n",
    "2. the httpx request is made with the kwargs\n",
    "3. `_build_response` to prepare as `Response` type\n",
    "4. In the `Response` is attribute `Response.parsed` which uses `_parse_response` to get response as json / dict\n",
    "\n",
    "The difference between with and sans `..._detailed()` is that with returns the whole response and sans only returns the parsed response. \n",
    "\n",
    "-----\n",
    "\n",
    "We will now make carry out the example request to get all studies of biome \"root:Host-associated:Plants:Rhizosphere\"."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d083a453",
   "metadata": {},
   "outputs": [],
   "source": [
    "# prep our search\n",
    "params = {\n",
    "    \"biome_name\": \"root:Host-associated:Plants:Rhizosphere\",\n",
    "    #this is default num of results per page in mgnify\n",
    "    \"page_size\": 25,\n",
    "    #\"search\": UNSET \n",
    "}\n",
    "\n",
    "# make the sync call and store respone\n",
    "with example_client as client:\n",
    "    response = studies_list.sync_detailed(\n",
    "        client=client, **params\n",
    "    )\n",
    "\n",
    "# check \n",
    "response.status_code"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c799096f",
   "metadata": {},
   "source": [
    "and if we take a look at the parsed content:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "599c46e3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b'{\"links\":{\"first\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies?biome_name=root%3AHost-associated%3APlants%3ARhizosphere&page=1&page_size=25\",\"last\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies?biome_name=root%3AHost-associated%3APlants%3ARhizosphere&page=6&page_size=25\",\"next\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies?biome_name=root%3AHost-associated%3APlants%3ARhizosphere&page=2&page_size=25\",\"prev\":null},\"data\":[{\"type\":\"studies\",\"id\":\"MGYS00006860\",\"attributes\":{\"accession\":\"MGYS00006860\",\"bioproject\":\"PRJEB102572\",\"samples-count\":5,\"is-private\":false,\"last-update\":\"2025-12-02T14:29:59\",\"secondary-accession\":\"ERP183963\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary data set PRJNA1303850\",\"study-name\":\"Metagenome assembly of PRJNA1303850 data set (Ecological Restoration and the Dominant Effects of Rhizosphere Microbes in Carbon Cycling in Mine Areas)\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860\"}},{\"type\":\"studies\",\"id\":\"MGYS00006231\",\"attributes\":{\"accession\":\"MGYS00006231\",\"bioproject\":\"PRJEB55060\",\"samples-count\":114,\"is-private\":false,\"last-update\":\"2025-07-05T06:19:56\",\"secondary-accession\":\"ERP139927\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA789467, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA789467 data set (Disentangling the genetic basis of rhizosphere microbiome assembly in tomato (metagenomics part)).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006231\"}},{\"type\":\"studies\",\"id\":\"MGYS00006810\",\"attributes\":{\"accession\":\"MGYS00006810\",\"bioproject\":\"PRJEB72764\",\"samples-count\":6,\"is-private\":false,\"last-update\":\"2025-05-29T19:18:52\",\"secondary-accession\":\"ERP157559\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA870599, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA870599 data set (metagenome sequencing during maize big trumpet period Raw sequence reads Metagenome).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006810\"}},{\"type\":\"studies\",\"id\":\"MGYS00006783\",\"attributes\":{\"accession\":\"MGYS00006783\",\"bioproject\":\"PRJEB72745\",\"samples-count\":8,\"is-private\":false,\"last-update\":\"2025-03-15T19:58:11\",\"secondary-accession\":\"ERP157540\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJEB57250, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere:Soil.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJEB57250 data set (Sharing of specific amino acids by bulk soil and Maize rhizosphere Prokaryotes).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006783\"}},{\"type\":\"studies\",\"id\":\"MGYS00006781\",\"attributes\":{\"accession\":\"MGYS00006781\",\"bioproject\":\"PRJEB72736\",\"samples-count\":8,\"is-private\":false,\"last-update\":\"2025-03-15T11:54:38\",\"secondary-accession\":\"ERP157518\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA821718, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA821718 data set (Investigation of the diversities and functions of maize native microbial community towards plant health management).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006781\"}},{\"type\":\"studies\",\"id\":\"MGYS00006777\",\"attributes\":{\"accession\":\"MGYS00006777\",\"bioproject\":\"PRJEB73514\",\"samples-count\":4,\"is-private\":false,\"last-update\":\"2025-03-14T23:14:44\",\"secondary-accession\":\"ERP158283\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA488251, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA488251 data set (rhizosphere metagenome Raw sequence reads).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006777\"}},{\"type\":\"studies\",\"id\":\"MGYS00006775\",\"attributes\":{\"accession\":\"MGYS00006775\",\"bioproject\":\"PRJEB72766\",\"samples-count\":5,\"is-private\":false,\"last-update\":\"2025-03-14T11:07:25\",\"secondary-accession\":\"ERP157562\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA607213, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA607213 data set (Maize rhizosphere metagenome Metagenome).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006775\"}},{\"type\":\"studies\",\"id\":\"MGYS00006770\",\"attributes\":{\"accession\":\"MGYS00006770\",\"bioproject\":\"PRJEB72765\",\"samples-count\":7,\"is-private\":false,\"last-update\":\"2025-03-13T22:55:43\",\"secondary-accession\":\"ERP157561\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA789701, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere:Soil.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA789701 data set (Taxonomic and functional diversity of rhizosphere microbiome recruited from compost synergistically determined by plant species and compost).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006770\"}},{\"type\":\"studies\",\"id\":\"MGYS00006766\",\"attributes\":{\"accession\":\"MGYS00006766\",\"bioproject\":\"PRJEB72718\",\"samples-count\":62,\"is-private\":false,\"last-update\":\"2025-03-13T14:37:08\",\"secondary-accession\":\"ERP157497\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA)  assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA728686, and was assembled with metaspades v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA728686 data set (Metatranscriptome analysis for rhizosphere soil from maize roots).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006766\"}},{\"type\":\"studies\",\"id\":\"MGYS00006652\",\"attributes\":{\"accession\":\"MGYS00006652\",\"bioproject\":\"PRJEB58189\",\"samples-count\":450,\"is-private\":false,\"last-update\":\"2025-02-21T18:00:13\",\"secondary-accession\":\"ERP143252\",\"centre-name\":\"Scotland\\'s Rural College - Aberdeen Campus\",\"public-release-date\":null,\"study-abstract\":\"The UK Crop Microbiome Cryobank (UK-CMCB) project (funded by the BBSRC (BB/T019484/1)) integrates genomic (DNA) data with a cryobank collection of samples for the soil microbiomes of six major UK crop plant systems. The UK crops were grown in a glasshouse environment in soils obtained from nine locations across the UK. The soils are agricultural in arable rotation systems, representative of different soil textural and taxonomic types. Microbiomes were collected from the crop rhizospheres at the developmental point of harvest. This study contains amplicon data (raw data in fastq files) for the V3-V4 region of bacterial 16S ribosomal gene and the ITS2 intergenic region of the fungal ribosomal genes, obtained from the rhizosphere of 6 crop plants (Fava Beans, Oilseed rape, Spring barley, Spring Oats, Spring Wheat, Sugar Beet) and bulk soil (Control).\",\"study-name\":\"UK Crop Microbiome CryoBank\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006652\"}},{\"type\":\"studies\",\"id\":\"MGYS00001368\",\"attributes\":{\"accession\":\"MGYS00001368\",\"bioproject\":\"PRJEB12501\",\"samples-count\":989,\"is-private\":false,\"last-update\":\"2024-03-22T19:03:54\",\"secondary-accession\":\"ERP013987\",\"centre-name\":\"Cornell University\",\"public-release-date\":null,\"study-abstract\":\"Soil management is necessary to ensure the fertility of reforested land, yet the long-term impact of harvesting on the soil microbial communities which mediate fertility remains largely a matter of conjecture. In 1989, the North American-wide \\xe2\\x80\\x98long-term soil productivity\\xe2\\x80\\x99 (LTSP) study was initiated to assess the impacts of organic matter removal on forest soils on a decadal timeframe. We sampled comprehensively within the LTSP framework in six biogeographic zones throughout British Columbia, California, Ontario and Texas, producing ITS and 16S pyrotag libraries. We observed major expansions in populations of desiccation, radiation and heat-resistant organisms from a broad diversity of fungal and bacterial groups. By identifying specific taxa, genes and processes altered post-harvesting, we contribute to refining long-term monitoring efforts and reveal important phenomenon impacting long-term forest regeneration.\",\"study-name\":\"Comprehensive amplicon-based metagenome analyses reveal long-term impacts of timber harvesting on soil microbial communities at reforested sites across North America.\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Forest soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Forest%20soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001368\"}},{\"type\":\"studies\",\"id\":\"MGYS00006578\",\"attributes\":{\"accession\":\"MGYS00006578\",\"bioproject\":\"PRJEB70816\",\"samples-count\":27,\"is-private\":false,\"last-update\":\"2024-02-15T10:42:46\",\"secondary-accession\":\"ERP155720\",\"centre-name\":\"UNISA\",\"public-release-date\":null,\"study-abstract\":\"Piante di nocciolo di 4 anni di due varieta\\' diverse (Tonda di Giffoni e Mortarella) concimate con fertilizzante di sintesi (NPK), compost di sansa, zolfo e bentonite, zolfo e bentonite + compost di sansa\",\"study-name\":\"Analisi metagenomica (16S) su suolo rizosferico di nocciolo\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006578\"}},{\"type\":\"studies\",\"id\":\"MGYS00000955\",\"attributes\":{\"accession\":\"MGYS00000955\",\"bioproject\":\"PRJNA247466\",\"samples-count\":36,\"is-private\":false,\"last-update\":\"2024-01-23T13:14:24\",\"secondary-accession\":\"SRP042949\",\"centre-name\":\"University of Munich\",\"public-release-date\":null,\"study-abstract\":\"Sequence arbuscular mycorrhizal fungi from soil samples from the Andean region\",\"study-name\":\"Arbuscular mycorrhizal fungi from soil samples Targeted Locus (Loci)\",\"data-origination\":\"HARVESTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000955\"}},{\"type\":\"studies\",\"id\":\"MGYS00000957\",\"attributes\":{\"accession\":\"MGYS00000957\",\"bioproject\":\"PRJNA251819\",\"samples-count\":70,\"is-private\":false,\"last-update\":\"2024-01-23T12:37:31\",\"secondary-accession\":\"SRP043030\",\"centre-name\":\"University of Montreal\",\"public-release-date\":null,\"study-abstract\":\"The aim of this study was to compare AMF communities from petroleum-contaminated and adjacent uncontaminated soils, and to determine whether willow planting influenced community composition.\",\"study-name\":\"AMF from contaminated and uncontaminated rhizosphere soils Metagenome\",\"data-origination\":\"HARVESTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000957\"}},{\"type\":\"studies\",\"id\":\"MGYS00001704\",\"attributes\":{\"accession\":\"MGYS00001704\",\"bioproject\":\"PRJEB20191\",\"samples-count\":383,\"is-private\":false,\"last-update\":\"2024-01-23T12:03:24\",\"secondary-accession\":\"ERP022327\",\"centre-name\":\"The Plant Phenotyping and Imaging Research Centre\",\"public-release-date\":null,\"study-abstract\":\"The study goal is to support crop breeders in identifying microbial communities and characteristics that improve crop yield. The microbial community associated with the roots of canola grown over a three-year period in a Saskatchewan field will be analyzed. DNA extraction and analysis results from the roots and soils will identify specific microbes or communities of microbes connected with the high yielding varieties. In combination with other phenotypic data, unique identifiers of the microbial community will be determined and then tested under controlled experimental conditions to validate their link in improved crop performance.\",\"study-name\":\"P2IRC 1.3 Soil microbiome as plant phenotype\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001704\"}},{\"type\":\"studies\",\"id\":\"MGYS00000867\",\"attributes\":{\"accession\":\"MGYS00000867\",\"bioproject\":\"PRJNA225620\",\"samples-count\":33,\"is-private\":false,\"last-update\":\"2024-01-23T11:51:44\",\"secondary-accession\":\"SRP032419\",\"centre-name\":\"The University of Arizona\",\"public-release-date\":null,\"study-abstract\":\"The objective of this study was to analyze the temporal dynamics of the microbial communities colonizing the rhizosphere of plants growing in compost-amended mine tailings with the goal of linking community structure to ecosystem function. Metalliferous mine tailings, which usually have high acid-generating potentials, are often subject to phytoremediation to prevent the dispersion (by eolian and hydraulic action) of these metal-bearing particles. Assisted phytostabilization is a type of phytoremediation technology in which the mine tailings are amended with compost (as a source of carbon and nutrients, and pH buffer) and then are seeded with a native plant specie that has been specifically selected based on its capacity to grow on the tailings without accumulating elevated levels of toxic metals in the shoot tissues. Therefore, the metal contaminants are immobilized in the soil to minimize the health and environmental risks associated with the dispersion of exposed mine tailings. A major knowledge gap in the understanding of phytostabilization as a viable alternative for the remediation of mine tailing piles is the ecology of the microorganisms that grow in the root-zone of the plants. Little is known of how these microbial communities respond to plant growth and compost amendment, how they may influence the stabilization of the metal contaminants, how they may affect or enhance plant colonization of the tailings, and how their structure may reflect the state of the phytoremediation treatment. To try to answer some of these questions, we conducted a yearlong greenhouse experiment with 3 treatments (in triplicate): (1) tailings only control, (2) tailings amended with 15% compost control, (3) tailings amended with 15% compost and seeded with a shrub (quail-bush; Atriplex lentiformis). During that year, core samples were collected every 3 months to extract metagenomic DNA and analyze the structure of the microbial communities (i.e. phylogenetic and functional structure).\",\"study-name\":\"Mesocosm mine tailing cores Targeted Locus (Loci)\",\"data-origination\":\"HARVESTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00000867\"}},{\"type\":\"studies\",\"id\":\"MGYS00006237\",\"attributes\":{\"accession\":\"MGYS00006237\",\"bioproject\":\"PRJEB55217\",\"samples-count\":12,\"is-private\":false,\"last-update\":\"2023-06-16T18:37:06\",\"secondary-accession\":\"ERP140100\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA726355, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA726355 data set (Responsiveness of rhizosphere bacteria assembled from manipulated soil microbiome suppressed Ralstonia solanacearum invasion).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006237\"}},{\"type\":\"studies\",\"id\":\"MGYS00006229\",\"attributes\":{\"accession\":\"MGYS00006229\",\"bioproject\":\"PRJEB55045\",\"samples-count\":12,\"is-private\":false,\"last-update\":\"2023-06-15T21:04:35\",\"secondary-accession\":\"ERP139910\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJEB30847, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJEB30847 data set (Shotgun metagenome sequence library from bulk soil and rhizosphere samples of wild and domesticated barley genotypes).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006229\"}},{\"type\":\"studies\",\"id\":\"MGYS00006232\",\"attributes\":{\"accession\":\"MGYS00006232\",\"bioproject\":\"PRJEB55074\",\"samples-count\":6,\"is-private\":false,\"last-update\":\"2023-06-15T15:22:44\",\"secondary-accession\":\"ERP139949\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA395704, and was assembled with megahit v1.2.9, metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metatranscriptomics assembly of PRJNA395704 data set (Metatranscriptome_Rhizosphere_Barley Transcriptome).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006232\"}},{\"type\":\"studies\",\"id\":\"MGYS00006230\",\"attributes\":{\"accession\":\"MGYS00006230\",\"bioproject\":\"PRJEB55057\",\"samples-count\":1,\"is-private\":false,\"last-update\":\"2023-06-15T13:46:24\",\"secondary-accession\":\"ERP139923\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA745270, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA745270 data set (metagenomics sequence of manure and unmanured rhizosphere soil of tomato plant).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006230\"}},{\"type\":\"studies\",\"id\":\"MGYS00006228\",\"attributes\":{\"accession\":\"MGYS00006228\",\"bioproject\":\"PRJEB54995\",\"samples-count\":1,\"is-private\":false,\"last-update\":\"2023-06-15T13:04:29\",\"secondary-accession\":\"ERP139862\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA763981, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA763981 data set (SOYBEAN RHIZOSPHERE METAGENOMICS).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006228\"}},{\"type\":\"studies\",\"id\":\"MGYS00006205\",\"attributes\":{\"accession\":\"MGYS00006205\",\"bioproject\":\"PRJEB55224\",\"samples-count\":12,\"is-private\":false,\"last-update\":\"2023-04-20T21:07:45\",\"secondary-accession\":\"ERP140107\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA755742, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA755742 data set (Tomato rhizosphere microbiome in the pot experiment).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006205\"}},{\"type\":\"studies\",\"id\":\"MGYS00006208\",\"attributes\":{\"accession\":\"MGYS00006208\",\"bioproject\":\"PRJEB55232\",\"samples-count\":1,\"is-private\":false,\"last-update\":\"2023-04-20T18:58:52\",\"secondary-accession\":\"ERP140115\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA766489, and was assembled with metaSPAdes v3.15.3, SPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA766489 data set (Culture-independent analysis of rhizosphere microbial communities associated with tomato plants).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006208\"}},{\"type\":\"studies\",\"id\":\"MGYS00006204\",\"attributes\":{\"accession\":\"MGYS00006204\",\"bioproject\":\"PRJEB55219\",\"samples-count\":12,\"is-private\":false,\"last-update\":\"2023-04-20T17:57:11\",\"secondary-accession\":\"ERP140102\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA755741, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA755741 data set (Tomato rhizosphere microbiome in the belowground segregation experiment).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006204\"}},{\"type\":\"studies\",\"id\":\"MGYS00006202\",\"attributes\":{\"accession\":\"MGYS00006202\",\"bioproject\":\"PRJEB55194\",\"samples-count\":17,\"is-private\":false,\"last-update\":\"2023-04-20T11:17:53\",\"secondary-accession\":\"ERP140076\",\"centre-name\":\"EMG\",\"public-release-date\":null,\"study-abstract\":\"The Third Party Annotation (TPA) assembly was derived from the primary whole genome shotgun (WGS) data set PRJNA603603, and was assembled with metaSPAdes v3.15.3. This project includes samples from the following biomes: root:Host-associated:Plants:Rhizosphere:Soil.\",\"study-name\":\"EMG produced TPA metagenomics assembly of PRJNA603603 data set (Soil and rhizospheric whole shotgun metagenomes).\",\"data-origination\":\"SUBMITTED\"},\"relationships\":{\"publications\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/publications\"}},\"downloads\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/downloads\"}},\"samples\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/samples\"}},\"biomes\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/biomes\"},\"data\":[{\"type\":\"biomes\",\"id\":\"root:Host-associated:Plants:Rhizosphere:Soil\",\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere:Soil\"}}]},\"analyses\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/analyses\"}},\"geocoordinates\":{\"links\":{\"related\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202/geocoordinates\"}}},\"links\":{\"self\":\"https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006202\"}}],\"meta\":{\"pagination\":{\"page\":1,\"pages\":6,\"count\":144}}}'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.parsed.to_dict().keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b1fea0d",
   "metadata": {},
   "source": [
    "- In `data` is the study metadata\n",
    "    - for the first page of results only\n",
    "- In `meta` is information on how many results and how many pages thaose results are spread out on based on the given `page_size`\n",
    "\n",
    "Note: you need to instantiate a new client object every call"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b826c6f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# TODO explore and get the number of samples from data\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "mgnipy",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
