{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ba92758e",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "from mgnipy_one.api.studies import studies_list\n",
    "from mgnipy_one import Client as ClientOne\n",
    "from mgnipy_one.models import PaginatedStudyList\n",
    "from mgnipy_one.types import Response\n",
    "\n",
    "base_url = \"https://www.ebi.ac.uk/metagenomics/api\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2dedf9c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "from mgnipy_one.types import UNSET"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49edf08c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Response [200 OK]>\n",
      "https://www.ebi.ac.uk/metagenomics/api/v1/studies?biome_name=root%3AHost-associated%3APlants%3ARhizosphere&page_size=100\n"
     ]
    }
   ],
   "source": [
    "biome_name = \"root:Host-associated:Plants:Rhizosphere\"\n",
    "search = UNSET\n",
    "page_size = 25 #default\n",
    "\n",
    "\n",
    "client = ClientOne(\n",
    "    base_url=base_url,\n",
    ")\n",
    "with client as client:\n",
    "    response: Response[PaginatedStudyList] = studies_list.sync_detailed(\n",
    "        client=client, biome_name=biome_name, search=search, page_size=page_size\n",
    "    )\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "47270898",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'type': 'studies',\n",
       " 'id': 'MGYS00006860',\n",
       " 'attributes': {'samples-count': 5,\n",
       "  'bioproject': 'PRJEB102572',\n",
       "  'accession': 'MGYS00006860',\n",
       "  'is-private': False,\n",
       "  'last-update': '2025-12-02T14:29:59',\n",
       "  'secondary-accession': 'ERP183963',\n",
       "  'centre-name': 'EMG',\n",
       "  'public-release-date': None,\n",
       "  'study-abstract': 'The Third Party Annotation (TPA) assembly was derived from the primary data set PRJNA1303850',\n",
       "  'study-name': 'Metagenome assembly of PRJNA1303850 data set (Ecological Restoration and the Dominant Effects of Rhizosphere Microbes in Carbon Cycling in Mine Areas)',\n",
       "  'data-origination': 'SUBMITTED'},\n",
       " 'relationships': {'analyses': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/analyses'}},\n",
       "  'publications': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/publications'}},\n",
       "  'biomes': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/biomes'},\n",
       "   'data': [{'type': 'biomes',\n",
       "     'id': 'root:Host-associated:Plants:Rhizosphere',\n",
       "     'links': {'self': 'https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Host-associated:Plants:Rhizosphere'}}]},\n",
       "  'geocoordinates': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/geocoordinates'}},\n",
       "  'samples': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/samples'}},\n",
       "  'downloads': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860/downloads'}}},\n",
       " 'links': {'self': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00006860'}}"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.parsed.to_dict()['data'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "ca961508",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_pages.to_dict()['meta']['pagination']['pages']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e06a8576",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'type': 'studies',\n",
       " 'id': 'MGYS00001560',\n",
       " 'attributes': {'samples-count': 2,\n",
       "  'bioproject': 'PRJEB6713',\n",
       "  'accession': 'MGYS00001560',\n",
       "  'is-private': False,\n",
       "  'last-update': '2024-04-15T16:21:47',\n",
       "  'secondary-accession': 'ERP006350',\n",
       "  'centre-name': 'CTBE',\n",
       "  'public-release-date': None,\n",
       "  'study-abstract': 'This study aimed to evaluate the long-term operation of an up-flow anaerobic packed bed reactor (APBR) filled with low-density of polyethylene (LDP) and applied to thermophilic hydrogen production using sugarcane vinasse as substrate. The APBR was operated at organic loading rate (OLR) of 84.2 kg-COD.m-3.d-1, value determined in earlier study. The maximum values of hydrogen production and yield were 5,252.6 mL-H2.d-1 and 3.7 mol-H2.mol-1total carbohydrates, respectively. However, as the OLR applied was constant, the values of specific organic load rate (sOLR) decreased throughout operation from values from 1.38 to 0.72 g-Total carbohydrates.g-VS-1.h-1, thus, interfering negatively on hydrogen production. sOLR values around 1.22 g-Total carbohydrates.g-VS-1.h-1 indicated optimal conditions for hydrogen production. The microbial community was study by 454-pyrosequencing analysis. Organisms belonging to genera Caloramotor, Clostridium, Thermoanaerobacterium, Thermohydrogenium, Caldanaerobius and Megasphaera were detected in samples taken from reactor at days 30 and 60 of operation, suggesting a role in the hydrogen production.',\n",
       "  'study-name': 'THERMOPHILIC HYDROGEN PRODUCTION IN ACIDOGENIC PACKED BED REACTOR (APBR) FED WITH SUGARCANE VINASSE: PERFORMANCE DURING LONG-TERM OPERATION',\n",
       "  'data-origination': 'SUBMITTED'},\n",
       " 'relationships': {'analyses': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/analyses'}},\n",
       "  'publications': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/publications'}},\n",
       "  'biomes': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/biomes'},\n",
       "   'data': [{'type': 'biomes',\n",
       "     'id': 'root:Engineered:Bioreactor:Continuous culture',\n",
       "     'links': {'self': 'https://www.ebi.ac.uk/metagenomics/api/v1/biomes/root:Engineered:Bioreactor:Continuous%20culture'}}]},\n",
       "  'geocoordinates': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/geocoordinates'}},\n",
       "  'samples': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/samples'}},\n",
       "  'downloads': {'links': {'related': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560/downloads'}}},\n",
       " 'links': {'self': 'https://www.ebi.ac.uk/metagenomics/api/v1/studies/MGYS00001560'}}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_pages.to_dict()['data'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "867db719",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<HTTPStatus.NOT_FOUND: 404>"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "response.status_code"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ab0a731",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4e1c156",
   "metadata": {},
   "outputs": [],
   "source": [
    "result = await studies_list.asyncio(\n",
    "    client=client1,\n",
    "    biome_name='root:Engineered:Biogas plant'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "81a03b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import tqdm.asyncio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "893d5a98",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "object of type 'coroutine' has no len()",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mTypeError\u001b[39m                                 Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[10]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[38;5;28;43;01mfor\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mf\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01min\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mtqdm\u001b[49m\u001b[43m.\u001b[49m\u001b[43masyncio\u001b[49m\u001b[43m.\u001b[49m\u001b[43mtqdm\u001b[49m\u001b[43m.\u001b[49m\u001b[43mas_completed\u001b[49m\u001b[43m(\u001b[49m\u001b[43mresult\u001b[49m\u001b[43m)\u001b[49m\u001b[43m:\u001b[49m\n\u001b[32m      2\u001b[39m \u001b[43m    \u001b[49m\u001b[38;5;28;43;01mawait\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mresult\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/github/mgnipy/.venv/lib/python3.11/site-packages/tqdm/asyncio.py:63\u001b[39m, in \u001b[36mtqdm_asyncio.as_completed\u001b[39m\u001b[34m(cls, fs, loop, timeout, total, **tqdm_kwargs)\u001b[39m\n\u001b[32m     59\u001b[39m \u001b[38;5;250m\u001b[39m\u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m     60\u001b[39m \u001b[33;03mWrapper for `asyncio.as_completed`.\u001b[39;00m\n\u001b[32m     61\u001b[39m \u001b[33;03m\"\"\"\u001b[39;00m\n\u001b[32m     62\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m total \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m---> \u001b[39m\u001b[32m63\u001b[39m     total = \u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mfs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     64\u001b[39m kwargs = {}\n\u001b[32m     65\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m version_info[:\u001b[32m2\u001b[39m] < (\u001b[32m3\u001b[39m, \u001b[32m10\u001b[39m):\n",
      "\u001b[31mTypeError\u001b[39m: object of type 'coroutine' has no len()"
     ]
    }
   ],
   "source": [
    "for f in tqdm.asyncio.tqdm.as_completed(result):\n",
    "    await result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b1644d1",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'coroutine' object has no attribute 'count'",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mAttributeError\u001b[39m                            Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[8]\u001b[39m\u001b[32m, line 1\u001b[39m\n\u001b[32m----> \u001b[39m\u001b[32m1\u001b[39m \u001b[43mresult\u001b[49m\u001b[43m.\u001b[49m\u001b[43mcount\u001b[49m\n",
      "\u001b[31mAttributeError\u001b[39m: 'coroutine' object has no attribute 'count'"
     ]
    }
   ],
   "source": [
    "await result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "743edd49",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
